### PR TITLE
Add fleet action results support.

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

* Fleet action results support.
* Fleet server integration test, running full server and currently covering unauthorized requests. Planning to incrementally build on top of that to cover the full agent life-cycle, hitting the actual API endpoints of the server.

### CODE REVIEW

**Comment by aleksmaus** [comment]
Well, failed integration test starting the server due to missing .kibana index, working on the fix.

**Comment by aleksmaus** [comment]
was cleaning up some old branches, deleted this one by mistake, thought it was merged. reopened.

File: cmd/fleet/router.go [hunk 1]
```diff
@@ -8,34 +8,39 @@ import (
...
+		ct:       ct,
+		et:       et,
+		validate: validator.New(),
 	}
 
 	router := httprouter.New()
 	router.POST(ROUTE_ENROLL, r.handleEnroll)
 	router.POST(ROUTE_CHECKIN, r.handleCheckin)
 	router.POST(ROUTE_ACKS, r.handleAcks)
+	router.POST(ROUTE_RESULT, r.handleActionResult)
```
Comment (blakerouse) [inline cmd/fleet/router.go:44]: Why the new endpoint? Why not just use the `ROUTE_ACKS`? Elastic Agent already handles actions and response with the Ack.

**Review by blakerouse — COMMENTED** [review]
Confused on the new endpoint? Why doesn't the `acks` endpoint work? Its already implemented to handle acking actions in Elastic Agent, no code change would be needed in the Elastic Agent for this.

File: cmd/fleet/router.go (hunk 1 above)
Comment (aleksmaus) [reply cmd/fleet/router.go:44]: It was a part of the original proposal doc and I didn't get any objections. The results endpoint is not an ack. The result document is posted upon completion of the action, with the details on of the completion an an error if error happened.
Assumed that the expiration/timeout of the action will be stored as the result document with the error code.

**Comment by aleksmaus** [comment]
With the new agents actions flow we are not acking actions as well. do acks endpoint is not used at least for actions.

File: cmd/fleet/router.go (hunk 1 above)
Comment (blakerouse) [reply cmd/fleet/router.go:44]: So agent will both ack and providing results?

Comment (aleksmaus) [reply cmd/fleet/router.go:44]: For the agent actions (that I worked on), the agent doesn't need to ack.
The workflow for actions looks something like this:
1. The Agent calls "checkin" API
2. The Fleet Server checked the "last action token"(if provided), and sends the new actions with the "token" in response. Next time the fleet server "checks in" it sends the token to the fleet server, this updates the agent record so we track the last action received by the agent.
3. The Agent dispatches the actions to the corresponding beats/applications
4. The beats/applications execute the action and stream the data to ES directly
5. The beats/applications communicate the result/error of the action execution back to the agent upon the completion of the action
6. The Agent calls the "result" API

**Review by blakerouse — APPROVED** [review]
Looks good.

Will take this and integrate it into the /acks

**Pull request merged**